packages/@aws-cdk/pipelines/lib/private/application-security-check.ts

```diff
@@ -0,0 +1,152 @@
+import * as path from 'path';
+import * as codebuild from '@aws-cdk/aws-codebuild';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import { Duration } from '@aws-cdk/core';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * A construct containing both the Lambda and CodeBuild Project
+ * needed to conduct a security check on any given application stage.
+ *
+ * The Lambda acts as an auto approving mechanism that should only be
+ * triggered when the CodeBuild Project registers no security changes.
+ *
+ * The CodeBuild Project runs a security diff on the application stage,
+ * and exports the link to the console of the project.
+ */
+export class ApplicationSecurityCheck extends Construct {
+  /**
+   * A lambda function that approves a Manual Approval Action, given
+   * the following payload:
+   *
+   * {
+   *  "PipelineName": [CodePipelineName],
+   *  "StageName": [CodePipelineStageName],
+   *  "ActionName": [ManualApprovalActionName]
+   * }
+   */
+  public readonly preApproveLambda: lambda.Function;
+  /**
+   * A CodeBuild Project that runs a security diff on the application stage.
+   *
+   * - If the diff registers no security changes, CodeBuild will invoke the
+   *   pre-approval lambda and approve the ManualApprovalAction.
+   * - If changes are detected, CodeBuild will exit into a ManualApprovalAction
+   */
+  public readonly cdkDiffProject: codebuild.Project;
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    this.preApproveLambda = new lambda.Function(this, 'CDKPipelinesAutoApprove', {
+      handler: 'index.handler',
+      runtime: lambda.Runtime.NODEJS_14_X,
+      code: lambda.Code.fromAsset(path.resolve(__dirname, '../lambda')),
+      timeout: Duration.minutes(5),
+    });
+
+    this.preApproveLambda.addToRolePolicy(new iam.PolicyStatement({
+      actions: ['codepipeline:GetPipelineState', 'codepipeline:PutApprovalResult'],
+      conditions: {
+        StringEquals: {
+          'aws:ResourceTag/SECURITY_CHECK': 'ALLOW_APPROVE',
+        },
+      },
+      resources: ['*'],
+    }));
+
+    const assemblyPath = 'assembly-$STACK_NAME-$STAGE_NAME/';
+    const invokeLambda =
+      'aws lambda invoke' +
+      ` --function-name ${this.preApproveLambda.functionName}` +
+      ' --invocation-type Event' +
+      ' --payload "$payload"' +
+      ' lambda.out';
+
+    const publishNotification =
+      'aws sns publish' +
+      ' --topic-arn $NOTIFICATION_ARN' +
+      ' --subject "$NOTIFICATION_SUBJECT"' +
+      ' --message "Changes detected! Requires Manual Approval.\n\n$LINK"';
```

Comment:
Better link to the pipeline; this won't link to the place where they have to click approve, and it'll be harder to find the right pipeline from the CodeBuild job (than the other way around).

Reply:
Hmm i think we should still link the CodeBuild job.. When a user gets the email "Hey you have security changes" I would want to be able to see those immediately.. I'll just add another one and label the two!